sync/src/block/extension.rs

```diff
@@ -339,13 +341,7 @@ impl NetworkExtension<Event> for Extension {
                         }
 
                         for id in peer_ids {
-                            let peer_score = if let Some(peer) = self.header_downloaders.get(&id) {
-                                peer.total_score()
-                            } else {
-                                U256::zero()
-                            };
-
-                            if peer_score > best_proposal_score {
+                            if self.header_downloaders.get(&id).map_or(false, |d| !d.is_lagging()) {
```

Comment:
`is_lagging` is not correct to use as a condition for `send_body_request`. When a node's peer has the same best block with the node, the node should not send a body request.

```rust
let is_lagging = {
  let peer_status = self.client.block_status(&BlockId::Hash(self.best_hash));
  let best_hash = self.client.block_hash(&BlockId::Latest).expect("Best hash always exists");
  peer_status == BlockStatus::InChain && self.best_hash != best_hash
}
```

correct condition
```rust
let is_lagging = {
  let peer_status = self.client.block_status(&BlockId::Hash(self.best_hash));
  let best_hash = self.client.block_hash(&BlockId::Latest).expect("Best hash always exists");
  peer_status == BlockStatus::InChain
}
```

Reply:
Thx. I fixed it.